CONTRIBUTING.md

```diff
@@ -1,5 +1,96 @@
 # Contributing to qiskit_ibm_transpiler
 
+## Release Notes
```

Comment:
This content is mostly copied from Qiskit Runtime, but it should be reviewed still. It's the new workflow your team would be expected to follow